ofrak_core/ofrak/core/ubifs.py

```diff
@@ -0,0 +1,176 @@
+import subprocess
+import tempfile
+from dataclasses import dataclass
+import logging
+
+from ofrak import Identifier, Analyzer
+from ofrak.component.packer import Packer
+from ofrak.component.unpacker import Unpacker
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File, Folder, FilesystemRoot, SpecialFileType
+from ofrak.core.binary import GenericBinary
+
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak_type.range import Range
+
+from ubireader.ubifs.defines import UBIFS_NODE_MAGIC
+
+from ubireader import ubi_io
+from ubireader.ubifs import ubifs as ubireader_ubifs
+from ubireader.ubifs.defines import PRINT_UBIFS_KEY_HASH, PRINT_UBIFS_COMPR
+from ubireader.utils import guess_leb_size
+
+LOGGER = logging.getLogger(__name__)
+
+MKFS_UBIFS_TOOL = ComponentExternalTool(
+    "mkfs.ubifs",
+    "http://www.linux-mtd.infradead.org/faq/ubifs.html",
+    install_check_arg="--help",
+    apt_package="mtd-utils",
+    brew_package="",  # This isn't compatible with macos, but there may be an alternative tool to do the bidding.
```

Comment:
```suggestion
    brew_package=None,  # This isn't compatible with macos, but there may be an alternative tool to do the bidding.
```
Should be set to None if we don't have one. See https://ofrak.com/docs/reference/ofrak/model/component_model.html#ofrak.model.component_model.ComponentExternalTool.